### PR TITLE
FileStatus names

### PR DESCRIPTION
**Pull request by nulltoken**
Current FileStatus names are missleading. To get know what they actually stands for, you have to read the doco. I like names that apear in ligbit2. When I see such name I know that state describes change made to INDEX or WorkingTree. 
We can drop `GIT_STATUS_` prefix as they are members of FileStatus.
I would only use `Unmodified` instead of `Current`

```

	GIT_STATUS_CURRENT = 0,

	GIT_STATUS_INDEX_NEW        = (1u << 0),
	GIT_STATUS_INDEX_MODIFIED   = (1u << 1),
	GIT_STATUS_INDEX_DELETED    = (1u << 2),
	GIT_STATUS_INDEX_RENAMED    = (1u << 3),
	GIT_STATUS_INDEX_TYPECHANGE = (1u << 4),

	GIT_STATUS_WT_NEW           = (1u << 7),
	GIT_STATUS_WT_MODIFIED      = (1u << 8),
	GIT_STATUS_WT_DELETED       = (1u << 9),
	GIT_STATUS_WT_TYPECHANGE    = (1u << 10),
	GIT_STATUS_WT_RENAMED       = (1u << 11),

	GIT_STATUS_IGNORED          = (1u << 14), 
```

### CODE REVIEW

**Comment by nulltoken** [comment]
Ready for review

**Comment by nulltoken** [comment]
@Therzok Very strange Mono failure in this build **[log](https://travis-ci.org/libgit2/libgit2sharp/jobs/61891390)**

```
Native stacktrace:
	mono() [0x4accac]
	/lib/x86_64-linux-gnu/libpthread.so.0(+0xfcb0) [0x7f5ec8d7fcb0]
	/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x35) [0x7f5ec89e6445]
	/lib/x86_64-linux-gnu/libc.so.6(abort+0x17b) [0x7f5ec89e9bab]
	/lib/x86_64-linux-gnu/libc.so.6(+0x73e2e) [0x7f5ec8a23e2e]
	/lib/x86_64-linux-gnu/libc.so.6(+0x7e626) [0x7f5ec8a2e626]
	mono() [0x53c411]
	mono() [0x61b4d8]
	mono() [0x61bc87]
	mono() [0x5dfffe]
	mono(mono_array_new_full+0x1e2) [0x5a6342]
	mono() [0x5a72f5]
	[0x40bfb8cb]
Debug info from gdb:
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================
./build.libgit2sharp.sh: line 34:  7743 Aborted                 mono Lib/NuGet/NuGet.exe restore LibGit2Sharp.sln
```

**Comment by Therzok** [comment]
Not sure what caused it. Seems to be mono 4. I think something got fixed regarding this in 4.1.

File: LibGit2Sharp/FileStatus.cs [hunk 1]
```diff
@@ -41,29 +59,59 @@ public enum FileStatus
...
 
         /// <summary>
+        /// The file type has been changed in the working directory. A previous version exists in the Index.
+        /// </summary>
+        TypeChangeInWorkdir = (1 << 10), /* GIT_STATUS_WT_TYPECHANGE */
+
+        /// <summary>
         /// The file has been renamed in the working directory.  The previous version at the previous name exists in the Index.
         /// </summary>
         RenamedInWorkDir = (1 << 11), /* GIT_STATUS_WT_RENAMED */
```
Comment (jbialobr) [inline LibGit2Sharp/FileStatus.cs]: It is a minor issue, but the all other names have 'dir' (lowercase d) while here is uppercased 'Dir'.

Comment (nulltoken) [inline LibGit2Sharp/FileStatus.cs]: @jbialobr Good catch! Fixed

**Comment by nulltoken** [comment]
Published as NuGet pre-release package `LibGit2Sharp.0.22.0-pre20150527180327`

**Pull request merged**